tests/Unit/Core/MailTemplate/FolderThemeCatalogTest.php

```diff
@@ -0,0 +1,375 @@
+<?php
+/**
+ * 2007-2019 PrestaShop SA and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace Tests\Unit\Core\MailTemplate;
+
+use PHPUnit\Framework\TestCase;
+use PrestaShop\PrestaShop\Core\Exception\FileNotFoundException;
+use PrestaShop\PrestaShop\Core\Hook\HookDispatcherInterface;
+use PrestaShop\PrestaShop\Core\MailTemplate\FolderThemeCatalog;
+use PrestaShop\PrestaShop\Core\MailTemplate\Layout\LayoutCollectionInterface;
+use PrestaShop\PrestaShop\Core\MailTemplate\Layout\LayoutInterface;
+use PrestaShop\PrestaShop\Core\MailTemplate\MailTemplateInterface;
+use PrestaShop\PrestaShop\Core\MailTemplate\Theme;
+use PrestaShop\PrestaShop\Core\MailTemplate\ThemeCatalogInterface;
+use PrestaShop\PrestaShop\Core\MailTemplate\ThemeCollection;
+use PrestaShop\PrestaShop\Core\MailTemplate\ThemeCollectionInterface;
+use PrestaShop\PrestaShop\Core\MailTemplate\ThemeInterface;
+use Symfony\Component\Filesystem\Filesystem;
+
+class FolderThemeCatalogTest extends TestCase
+{
+    /**
+     * @var string
+     */
+    private $tempDir;
+
+    /**
+     * @var Filesystem
+     */
+    private $fs;
+
+    /**
+     * @var ThemeCollectionInterface
+     */
+    private $expectedThemes;
+
+    /**
+     * @var array
+     */
+    private $coreLayouts;
+
+    /**
+     * @var array
+     */
+    private $moduleLayouts;
+
+    public function setUp()
+    {
+        parent::setUp();
+        $this->tempDir = sys_get_temp_dir() . DIRECTORY_SEPARATOR . 'mail_layouts';
```

Comment:
Hi ! I know I'm late on this one 😅 however it would have been cool to use https://github.com/mikey179/vfsStream as we did in https://github.com/PrestaShop/PrestaShop/blob/develop/tests/Unit/Core/Util/Url/UrlFileCheckerTest.php in order to avoid a hard-dependency on the filesystem. This might cause the tests to fail in some environments.

Reply:
Hi! There are other issues for this feature, we can integrate it in the next ones. I'll try to remember it ^^